worlds/timespinner/__init__.py

```diff
@@ -75,56 +77,82 @@ def create_items(self) -> None:
 
     def set_rules(self) -> None:
         final_boss: str
-        if self.is_option_enabled("DadPercent"):
+        if self.options.dad_percent:
             final_boss = "Killed Emperor"
         else:
             final_boss = "Killed Nightmare"
 
         self.multiworld.completion_condition[self.player] = lambda state: state.has(final_boss, self.player) 
 
     def fill_slot_data(self) -> Dict[str, object]:
-        slot_data: Dict[str, object] = {}
-
-        ap_specific_settings: Set[str] = {"RisingTidesOverrides", "TrapChance"}
-
-        for option_name in timespinner_options:
-            if (option_name not in ap_specific_settings):
-                slot_data[option_name] = self.get_option_value(option_name)
-
-        slot_data["StinkyMaw"] = True
-        slot_data["ProgressiveVerticalMovement"] = False
-        slot_data["ProgressiveKeycards"] = False
-        slot_data["PersonalItems"] = self.get_personal_items()
-        slot_data["PyramidKeysGate"] = self.precalculated_weights.pyramid_keys_unlock
-        slot_data["PresentGate"] = self.precalculated_weights.present_key_unlock
-        slot_data["PastGate"] = self.precalculated_weights.past_key_unlock
-        slot_data["TimeGate"] = self.precalculated_weights.time_key_unlock
-        slot_data["Basement"] = int(self.precalculated_weights.flood_basement) + \
-                                int(self.precalculated_weights.flood_basement_high)
-        slot_data["Xarion"] = self.precalculated_weights.flood_xarion
-        slot_data["Maw"] = self.precalculated_weights.flood_maw
-        slot_data["PyramidShaft"] = self.precalculated_weights.flood_pyramid_shaft
-        slot_data["BackPyramid"] = self.precalculated_weights.flood_pyramid_back
-        slot_data["CastleMoat"] = self.precalculated_weights.flood_moat
-        slot_data["CastleCourtyard"] = self.precalculated_weights.flood_courtyard
-        slot_data["LakeDesolation"] = self.precalculated_weights.flood_lake_desolation
-        slot_data["DryLakeSerene"] = not self.precalculated_weights.flood_lake_serene
-        slot_data["LakeSereneBridge"] = self.precalculated_weights.flood_lake_serene_bridge
-        slot_data["Lab"] = self.precalculated_weights.flood_lab
-
-        return slot_data
+        return {
```

Comment:
```suggestion
		options = self.options.as_dict("start_with_jewelry_box", "downloadable_items", <etc>, casing="pascal")
```

Reply:
Id rather not do this, it wont add much to readability

Reply:
it's a lot more readable if you put the option names on their own lines

Reply:
oke i will keep it the way it is now

Reply:
I'm seconding this idea FWIW